src/components/pages/Navbar/Navbar.js

```diff
@@ -116,80 +105,78 @@ const Navbar = ({ userProfile, getProfile, currentUser }) => {
-                      >
-                        <div className="username">
-                          Welcome {userProfile.first_name}
+  if (!user) {
+    return <NavBarLanding />;
+  } else {
+    return (
+      <>
+        <Layout className="layout">
+          <Header className="menuBar">
+            <div className="logoDiv">
+              <div onClick={reloadLogo}>
+                <img
+                  src={logo}
+                  alt="underdog devs logo"
+                  height="68"
+                  style={{ marginLeft: '1vw' }}
+                  role="button"
+                />
+              </div>
+              {isMentor && (
+                <Popover
+                  title={`Status: ${
+                    toggleStatus ? 'Accepting' : 'Not Accepting'
+                  }`}
+                  content={<MentorPopover />}
+                  placement="bottom"
+                >
+                  <section className="mentorStatus">
+                    <Switch
+                      checked={toggleStatus}
+                      onChange={handleToggleChange}
+                      id="mentorSwitch"
+                    />
+                    <span className="toggleText">New Mentees</span>
+                  </section>
+                </Popover>
+              )}
+              {Object.keys(user).length && (
+                <div className="userInfo-and-profilePic">
+                  <Dropdown overlay={accountMenu} placement="bottom" arrow>
+                    <div className="userInfo-and-profilePic">
+                      <div className="userInfo">
+                        <div className="username" aria-label="Account settings">
+                          <div className="username">
+                            Welcome {userProfile.first_name}
+                          </div>
                         </div>
                       </div>
                     </div>
-                  </div>
-                </Dropdown>
-              </div>
-            )}
-            {!isAuthenticated && (
-              <div className="header_buttons">
-                <LoginButton />
-                <ApplyButton />
-                <LogoutButton />
-              </div>
-            )}
-            {/* temporary logout button until private route is finished and when we can logout from dashboard */}
-          </div>
-        </Header>
-      </Layout>
-      <Modal
-        visible={modal}
-        onCancel={cancelOpen}
-        onOk={handleLogout}
-        title="Confirm Log Out"
-        role="logout"
-      >
-        Are you sure you want to log out now?
-      </Modal>
-    </>
-  );
+                  </Dropdown>
+                </div>
+              )}
+              {!isAuthenticated && (
+                <div className="header_buttons">
+                  <LoginButton />
+                  <ApplyButton />
+                </div>
+              )}
+              <LogoutButton />
```

Comment:
@ryan-hamblin it turns out that all 3 of those button components perform their own checks using isAuthenticated. However, on page load isAuthenticated is always false at first, so currently we always see the login and apply buttons for a moment or two before isAuthenticated becomes true and the logout button displays. I think a cleaner way to do this may be to move the logout button back into that div and replace isAuthenticated with isLoading....that way the buttons don't attempt to load until isAuthenticated is established, and then we would only see the correct buttons.
```suggestion
              const { logout, isAuthenticated, isLoading } = useAuth0();
              ...
              {!isLoading && (
                <div className="header_buttons">
                  <LoginButton />
                  <ApplyButton />
                  <LogoutButton />
                </div>
              )}
              
```

Reply:
This seems like a good suggestion from @tim-sansone to improve the UX by preventing those irrelevant and potentially confusing buttons from appearing momentarily before the correct button is displayed.